acceptance/tests/ticket_1429_facter_with_strict_flag.rb

```diff
@@ -0,0 +1,11 @@
+test_name 'ticket FACT-1429  facter should return exit code 1 for querying non-existing fact when --strict is on'
+
+agents.each do |host|
+  step 'facter should return exit code 0 for querying non-existing-fact without --strict flag'
+  on(host, facter('-p non-existing-fact'))
+
+  step 'facter should return exit code 1 for querying non-existing-fact with --strict flag'
+  on(host, facter('-p non-existing-fact --strict'), :acceptable_exit_codes => 1) do
+    assert_match(/ERROR puppetlabs.facter - fact "non-existing-fact" does not exist/, stderr.chomp, 'Unexpected error was detected!')
```

Comment:
yes, it works with the assert_match, but it's better if we check the result.exit_code that way, will update in the next push